python/hsfs/feature_view.py

```diff
@@ -3590,28 +3593,36 @@ def enable_logging(self) -> None:
         # Raises
             `hsfs.client.exceptions.RestAPIError` in case the backend fails to enable feature logging.
         """
-        return self._feature_view_engine.enable_feature_logging(self)
+        fv = self._feature_view_engine.enable_feature_logging(self)
+        self._feature_logging = self._feature_view_engine.get_feature_logging(fv)
+        return fv
 
     def log(
         self,
-        features: Union[pd.DataFrame, list[list], np.ndarray],
+        features: Union[
+            pd.DataFrame, list[list], np.ndarray, TypeVar("pyspark.sql.DataFrame")
+        ],
         predictions: Optional[Union[pd.DataFrame, list[list], np.ndarray]] = None,
         transformed: Optional[bool] = False,
         write_options: Optional[Dict[str, Any]] = None,
         training_dataset_version: Optional[int] = None,
-        hsml_model=None,
-    ):
-        """Log features and optionally predictions for the current feature view.
+        hopsworks_model=None,
```

Comment:
Do we need to specify `hopsworks_`? We could use just `model`, and `if not isinstance(model, hopsworks.model.Model)` we raise a ValueError("Model must be a hopsworks model").
Then the API looks cleaner